TM-10 Routing

TomsBemis opened the pull request:
Create task, Task list and Task details components moved to separate pages. Navigation links from task item to task details and to home page. New attribute 'id' for Task model and generator service for id number.
**Links to routes on the app in the anchor elements:**
<img width="720" alt="TM-10 PR 1" src="https://github.com/TomsBemis/task-manager-frontend/assets/149372128/922a2c42-663d-49b1-9dd8-53f1ed99da23">
**New task details page accessed by passing task id as URL parameter:**
<img width="747" alt="TM-10 PR 2" src="https://github.com/TomsBemis/task-manager-frontend/assets/149372128/83804d70-2d3c-4440-a5f6-604bceeafcc6">
**Create task component in separate page accessed through specific route:**
<img width="741" alt="TM-10 PR 3" src="https://github.com/TomsBemis/task-manager-frontend/assets/149372128/b74f3e10-41cc-40c1-8239-ac1a507e5348">

TomsBemis committed:
TM-10 Routing

Create task, Task list and Task details components moved to separate pages. Navigation links from task item to task details and to home page. New attribute 'id' for Task model and generator service for id number.

wanpii commented on src/app/shared/nav/nav.component.html:
```diff
@@ -0,0 +1,13 @@
+    <a class="navbar-brand" routerLink="/">Task Manager</a>
+    <ul class="navbar-nav">
+      <li class="nav-item">
+        <a class="nav-link" [routerLinkActiveOptions]="{exact: true}" routerLinkActive="active" routerLink="/tasks">Task List</a>
```
Do you need routerLinkActive and routerLInkOptions? It should work without them I think, please check :)

wanpii commented on src/app/tasks/task-details/task-details.component.ts:
```diff
@@ -0,0 +1,30 @@
+  task: Task | null = null;
+
+  onDeleted() {
+    if(this.task !== null && this.task.id !== null) {
```
instead of using this.task you can do this check in the template if ( task !== null) etc etc and then you can directly pass the task object or task id, since you only need id through the template to the method, that way you wont have to do these checks here and it'll be a little cleaner

wanpii commented on src/app/tasks/task.model.ts:
```diff
@@ -5,6 +5,7 @@ export type Option = {
 }
 
 export interface Task {
+    id: number | null,
```
I don't think ID will ever be null, when you implement BE each task item will have a generated ID

TomsBemis committed:
TM-10 Pull Request 4 requested changes 1

Task id attribute only allows number type. Task details template passes the id of the task to the method. Unnecessary routerLink directives removed.
